cg/meta/archive/ddn.py

```diff
@@ -0,0 +1,141 @@
+        response: Response = APIRequest.api_request_from_content(
+            api_method=APIMethods.POST,
+            url=urljoin(base=self.url, url="auth/token/refresh"),
+            headers=self.headers,
+            json={
+                "refresh": self.refresh_token,
+            },
+        )
+        response_content: dict = ReadStream.get_content_from_stream(
+            file_format=FileFormat.JSON, stream=response.content
+        )
+        self.auth_token: str = response_content.get("access")
+        self.token_expiration: datetime = datetime.fromtimestamp(response_content.get("expire"))
+
+    @property
+    def auth_header(self) -> Dict[str, str]:
+        """Returns an authorisation header based on the current auth token, or updates it if
+        needed."""
+        if datetime.now() > self.token_expiration:
+            self._refresh_auth_token()
+        return {"Authorization": f"Bearer {self.auth_token}"}
+
+    def archive_folders(self, sources_and_destinations: Dict[Path, Path]) -> bool:
+        """Archives all folders provided, to their corresponding destination, as given by the
+        dict."""
+        payload: dict = {
+            "pathInfo": self._format_paths_archive(
+                sources_and_destinations=sources_and_destinations
+            ),
+            "osType": self.ostype,
+            "createFolder": True,
+            "metadataList": [],
+        }
+        response: Response = APIRequest.api_request_from_content(
+            api_method=APIMethods.POST,
+            url=urljoin(base=self.url, url="files/archive"),
+            headers=dict(self.headers, **self.auth_header),
+            json=payload,
+        )
+        print(payload)
+        return response
+
+    def retrieve_folders(self, sources_and_destinations: Dict[Path, Path]) -> bool:
+        """Retrieves all folders provided, to their corresponding destination, as given by the
+        dict."""
+        payload: dict = {
+            "pathInfo": self._format_paths_retrieve(
+                sources_and_destinations=sources_and_destinations
+            ),
+            "osType": self.ostype,
+            "createFolder": True,
+        }
+        response: Response = APIRequest.api_request_from_content(
+            api_method=APIMethods.POST,
+            url=urljoin(base=self.url, url="files/retrieve"),
+            headers=dict(self.headers, **self.auth_header),
+            json=payload,
+        )
+        return response.ok
+
+    def _format_paths_archive(
+        self, sources_and_destinations: Dict[Path, Path]
+    ) -> List[Dict[str, str]]:
+        """Formats the given archiving-dictionary into the data structure specified by the
+        REST-API."""
+        return [
+            {
+                "source": Path(self.local_storage, source.relative_to("/home")).as_posix(),
+                "destination": self.archive_repository + destination.as_posix(),
```

Comment:
From the perspective of DDN and the datamovers the correct paths have `/proj` as their root dir, forcing the removal of `/home` if we want to pass the absolute path from hasta (which I think we should).

However I could not make up my mind regarding where to do this trimming. Open for other opinions here as well.